thoth/common/openshift.py

```diff
@@ -1479,19 +1488,37 @@ def schedule_adviser(
             raise ConfigurationError(
                 "Unable to schedule adviser without backend namespace being set"
             )
-
-        job_id = job_id or self.generate_id("adviser")
-        parameters = locals()
-        parameters.pop("self", None)
-        return self._schedule_workload(
-            run_method_name=self.run_adviser.__name__,
-            run_method_parameters=parameters,
-            template_method_name=self.get_adviser_template.__name__,
-            job_id=job_id,
-            namespace=self.backend_namespace,
-            labels={"component": "adviser"},
+        adviser_id = job_id or self.generate_id("adviser")
+        template_parameters = {}
+        template_parameters["THOTH_ADVISER_JOB_ID"] = adviser_id
+        template_parameters["THOTH_ADVISER_REQUIREMENTS"] = application_stack["requirements"]
+        template_parameters["THOTH_ADVISER_REQUIREMENTS_LOCKED"] = application_stack["requirements_lock"]
+        template_parameters["THOTH_ADVISER_LIBRARY_USAGE"] = json.dumps(library_usage)
+        template_parameters["THOTH_ADVISER_REQUIREMENTS_FORMAT"] = "pipenv"
+        template_parameters["THOTH_ADVISER_RECOMMENDATION_TYPE"] = recommendation_type
+        template_parameters["THOTH_ADVISER_RUNTIME_ENVIRONMENT"] = runtime_environment
+        template_parameters["THOTH_ADVISER_LIMIT"] = limit
+        template_parameters["THOTH_ADVISER_COUNT"] = count
+        template_parameters["THOTH_ADVISER_LIMIT_LATEST_VERSIONS"] = limit_latest_versions
+
+        workflow_parameters = {}
+        workflow_parameters["THOTH_DOCUMENT_ID"] = f"advise-{adviser_id}"
+
+        workflow_manager = WorkflowManager(
+            ocp_config={
+                "kubernetes_verify_tls": self.kubernetes_verify_tls
+                }
         )
 
+        return self._schedule_workflow(
```

Comment:
Can we introduce a new environment variable `THOTH_USE_ARGO`, which would (if set to `1`) cause triggering the new implementation in thoth-common that uses argo? If set to `0` (or not set), the old implementation that uses our workload-operator+graph-sync-scheduler would be used. This way we can transparently switch and/or fallback to the old implementation until things stabilize.